src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs

```diff
@@ -1352,6 +1362,76 @@ public void Save(ModelSaveContext ctx)
 
         public IDataView Read(string path) => Read(new MultiFileSource(path));
 
+        internal static TextLoader CreateTextReader<TInput>(IHostEnvironment host,
+           bool hasHeader = DefaultArguments.HasHeader,
+           char separator = DefaultArguments.Separator,
+           bool allowQuotedStrings = DefaultArguments.AllowQuoting,
+           bool supportSparse = DefaultArguments.AllowSparse,
+           bool trimWhitespace = DefaultArguments.TrimWhitespace)
+        {
+            var userType = typeof(TInput);
+
+            var fieldInfos = userType.GetFields(BindingFlags.Public | BindingFlags.Instance);
+
+            var propertyInfos =
+                userType
+                .GetProperties(BindingFlags.Public | BindingFlags.Instance)
+                .Where(x => x.CanRead && x.CanWrite && x.GetGetMethod() != null && x.GetSetMethod() != null && x.GetIndexParameters().Length == 0);
+
+            var memberInfos = (fieldInfos as IEnumerable<MemberInfo>).Concat(propertyInfos).ToArray();
+
+            var columns = new List<Column>();
+
+            for (int index = 0; index < memberInfos.Length; index++)
+            {
+                var memberInfo = memberInfos[index];
+                var mappingAttr = memberInfo.GetCustomAttribute<LoadColumnAttribute>();
+
+                if(mappingAttr == null)
+                    continue;
```

Comment:
hmm, no longer throwing? I'm not sure I'm a fan of it.

Reply:
@tomfinley @glebuk what do you think?

---
In reply to: [243120594](https://github.com/dotnet/machinelearning/pull/1878#discussion_r243120594) [](ancestors = 243120594)

Reply:
I agree with you Senja, is there some benefit with throwing?

---
In reply to: [243358643](https://github.com/dotnet/machinelearning/pull/1878#discussion_r243358643) [](ancestors = 243358643,243120594)

Reply:
> If we throw, we restrict usage by expecting the class to be dedicated to just loading the data, whereas if we don't the type can be used for other thing, and the decorated fields can be used for loading.

Interesting @sfilipi. So, it strikes me that there are two possible dangers we must guard against.

1. A user attempts to use a class to define a schema, *without* proper attribution by mistake. We know this happens a lot.
2. A user attempts to use a class to define a schema, *with* proper attribution, but wants to have some additional "extra" properties free for no particular reason.

Like I said I know 1 comes up, and throwing has helped people to guard against misuse. 2 has been the state for a while now. In fact, if we consider all prior usage of the API (where we follow this pattern), and all prior usage of the text loader, and I think of all the questions that we've gotten *about* them over the years, both outside of Microsoft in ML.NET and inside Microsoft with our internal tool, I don't recall that ever happening. (Have we received requests for this? I am finding it difficult to imagine a concrete scenario.)

Even if the scenario **were** somehow to come up, I would prefer to solve it by introducing another attribute to indicate that this property is not to be considered by the loader, rather than to have this behavior. So, something like, `[NoLoader]` or whatnot. (I don't insist on the name. I also don't want us to add it now, until there's a concrete request for it, since my suspicion is that it will never come up.)

For this reason, I disagree @sfilipi and @singlis and agree with @Zruty0. It must throw.